conjure/objectwriter.go

```diff
@@ -28,10 +29,28 @@ const (
 	dataVarName     = "data"
 )
 
+// To avoid re-computing safety recursively, we keep the computed result in this global cache.
+var safetyCache = make(map[types.Type]spec.LogSafety)
+
 func writeObjectType(file *jen.Group, objectDef *types.ObjectType) {
 	// Declare struct type with fields
 	containsCollection := false // If contains collection, we need JSON methods to initialize empty values.
-	file.Add(objectDef.Docs.CommentLine()).Type().Id(objectDef.Name).StructFunc(func(structDecl *jen.Group) {
+
+	// Compute the overall safety of the struct and add comment-based annotation
+	overallSafety := computeStructSafety(objectDef.Fields)
+	var safetyComment jen.Code = jen.Null()
```

Comment:
nit, but I think that `jen.Null` and `nil` are handled in the same manner, so I think it might be a bit more canonical to just have `var safetyComment jen.Code`